onnx/defs/controlflow/defs.cc

```diff
@@ -277,9 +277,18 @@ void ClearShape(TypeProto& input_type) {
 
 void LoopInferenceFunction(InferenceContext& ctx) {
   auto num_inputs = ctx.getNumInputs();
+  auto num_outputs = ctx.getNumOutputs();
   assert(num_inputs >= 2);
   auto num_loop_state_vars = num_inputs - 2; // skip 'M' and 'cond'
 
+  if (num_loop_state_vars > num_outputs) {
+    fail_type_inference(
+        "The number of loop state variables for input is larger than the number of outputs. ",
+        num_loop_state_vars,
+        " > ",
+        num_outputs);
+  }
+
```

Comment:
IIUC, we cannot get the exact number of loop stare variable in outputs. (don't know the number of scan_outputs K either) So I only made a check to make sure that at least the number of output cannot be smaller than the number of loop stare variable in input. @gramalingam please review it. If it looks good to you, I will add it in older Loop inference function as well.

Another question: do we need to do similar check for Scan op?

Reply:
We can figure out the number of scan-outputs from the number of outputs of the loop-body subgraph.

Reply:
I think we already have the necessary check in line 342? So, may be the check you added for variadic part is sufficient?